tools/firmware-utils/src/tplink-safeloader.c

```diff
@@ -278,6 +278,45 @@ static struct device_info boards[] = {
 		.first_sysupgrade_partition = "os-image",
 		.last_sysupgrade_partition = "support-list",
 	},
+	/** Firmware layout for the CPE610V1 */
+	{
+		.id     = "CPE610V1",
+		.vendor = "CPE610(TP-LINK|UN|N300-5|00000000):1.0\r\n",
+		.support_list =
+			"SupportList:\r\n"
+			"CPE610(TP-LINK|EU|N300-5|00000000):1.0\r\n"
+			"CPE610(TP-LINK|EU|N300-5|45550000):1.0\r\n"
+			"CPE610(TP-LINK|EU|N300-5|55530000):1.0\r\n"
+			"CPE610(TP-LINK|UN|N300-5|00000000):1.0\r\n"
+			"CPE610(TP-LINK|UN|N300-5|45550000):1.0\r\n"
+			"CPE610(TP-LINK|UN|N300-5|55530000):1.0\r\n"
+			"CPE610(TP-LINK|US|N300-5|55530000):1.0\r\n"
+			"CPE610(TP-LINK|UN|N300-5):1.0\r\n"
+			"CPE610(TP-LINK|EU|N300-5):1.0\r\n"
+			"CPE610(TP-LINK|US|N300-5):1.0\r\n",
+		.support_trail = '\xff',
+		.soft_ver = NULL,
+
+		.partitions = {
+			{"fs-uboot", 0x00000, 0x20000},
+			{"partition-table", 0x20000, 0x02000},
+			{"default-mac", 0x30000, 0x00020},
+			{"product-info", 0x31100, 0x00100},
+			{"signature", 0x32000, 0x00400},
+			{"os-image", 0x40000, 0x1c0000},
```

Comment:
Here you should use dynamic partitions so that it will be autogenerated and adjusted when building.
Example:
https://github.com/openwrt/openwrt/commit/a2a972b2cfd762bc2207ffddde228737245ab926#diff-b232b41a358a2b44c64fb32925bd265a

Reply:
This PR uses the OKLI-loader, which the linked commit doesn't.
Will the OKLI-loader interfere with dynamic partitioning?
If not, changes should be easy...

Reply:
It should not interfere.
Why is okli even used?

Reply:
Just put
{"firmware", 0x40000, 0x770000},
instead of os-image and file-system

Do NOT change first_/last_sysupgrade_partition

Reply:
I tried it and it gets stuck in a bootloop